### PR TITLE
Refactored the SEO plugin to make it extensible and error-proof

### LINKED ISSUE #7107: Provide a way to specify an alternate whois provider - SEO plugin
Requested by a user in #7103 

We could post an event in `SEO.getRank` https://github.com/piwik/piwik/blob/2.11.0-b3/plugins/SEO/API.php#L41 to initialize the `$data` array. We would not overwrite any existing keys / ranks afterwards.

We could alternatively or on top add plugin settings to disable/enable each provider. Eg if someone is not interested in whois or Alexa or Google you would simply uncheck it (SystemSetting I reckon)

### LINKED ISSUE #7153: The SEO widget should be displayed even if when a http service calls fail
When building the SEO widget, we collect metrics from various web services eg. Google, microsoft, whois, etc. In some cases the HTTP calls can fail. For example Whois may be rate limited to 100 requests per hour. 

Possible steps of this issue
 1.  investigate what possible limits are, check whether we can avoid them
 2. check that errors are logged (as a warning maybe?) 
 3. that the SEO widget is displayed with the failed value as `-` or `Network error` (and other values displayed correctly).

follows up https://github.com/piwik/piwik/issues/7103

### PR DESCRIPTION
**Pull request by mnapoli**
- Fixes #7153: errors while connecting to a 3rd party service are now logged as warning, and shown in the widget as `-` instead of the values
- Fixes #7107: a new event `SEO.getMetricsProviders` allows other plugins to register `MetricsProvider`

Also kinda related to #4172: as I said there I think we could move our implementations (that are now more decoupled) into a new Piwik "SEO" component.

### CODE REVIEW

File: plugins/SEO/Metric/Alexa.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?php
+
```
Comment (mattab) [inline plugins/SEO/Metric/Alexa.php]: * piwik file headers missing in some of the new files

**Comment by mattab** [comment]
Very nice refactor! It would be nice to create a new component for SEO metrics, feel free to do it quick if you like the idea :-)

**Comment by mnapoli** [comment]
OK!

File: plugins/SEO/Metric/Alexa.php (hunk 1 above)
Comment (mnapoli) [inline plugins/SEO/Metric/Alexa.php]: good catch ;)

**Comment by mnapoli** [comment]
By the way, by OK I meant I'll look into it later but not in this PR especially since we need it for 2.11 ;)

So ready to be merged if all OK

**Comment by mattab** [comment]
Looks good, I'll test it on demo2 and demo!

**Pull request merged**